src/server/search/doc_index.h

```diff
@@ -52,7 +52,102 @@ struct SearchResult {
   std::optional<facade::ErrorReply> error;
 };
 
-template <typename T> using SearchField = std::pair<T /*identifier*/, T /*short name*/>;
+enum class NameType : uint8_t { kIdentifier, kShortName, kUndefined };
+
+template <typename T> class SearchField {
```

Comment:
Please add comments what problem this type solves, that new_alias is... I.e. combining access to indentifier, alias lookup and new passed alias in one place